uniffi_bindgen/src/bindings/swift/templates/RustBufferHelper.swift

```diff
@@ -225,6 +225,75 @@ extension Bool: ViaFfi {
     }
 }
 
+extension Date: ViaFfiUsingByteBuffer, ViaFfi {
+    fileprivate static func read(from buf: Reader) throws -> Self {
+        let seconds: Int64 = try buf.readInt()
+        let microseconds: UInt32 = try buf.readInt()
+        if seconds >= 0 {
+            let delta = Double(seconds) + (Double(microseconds) / 1.0e9)
+            return Date.init(timeIntervalSince1970: delta)
+        } else {
+            let delta = Double(-seconds) + (Double(microseconds) / 1.0e9)
+            return Date.init(timeIntervalSince1970: -delta)
+        }
+    }
+
+    fileprivate func write(into buf: Writer) {
+        var delta = self.timeIntervalSince1970
+        var sign: Int64 = 1
+        if delta < 0 {
+            sign = -1
+            delta = -delta
+        }
+        if delta.rounded(.down) > Double(Int64.max) {
+            fatalError("Timestamp overflow, exceeds max bounds supported by Uniffi")
+        }
+        let seconds = Int64(delta)
+        let microseconds = UInt32((delta - Double(seconds)) * 1.0e9)
+        buf.writeInt(sign * seconds)
+        buf.writeInt(microseconds)
+    }
+}
+
+extension TimeInterval {
+    fileprivate static func liftDuration(_ buf: RustBuffer) throws -> Self {
+      let reader = Reader(data: Data(rustBuffer: buf))
+      let value = try Self.readDuration(from: reader)
+      if reader.hasRemaining() {
+          throw UniffiInternalError.incompleteData
+      }
+      buf.deallocate()
+      return value
+    }
+
+    fileprivate func lowerDuration() -> RustBuffer {
+      let writer = Writer()
+      self.writeDuration(into: writer)
+      return RustBuffer(bytes: writer.bytes)
+    }
+
+    fileprivate static func readDuration(from buf: Reader) throws -> Self {
+        let seconds: UInt64 = try buf.readInt()
+        let microseconds: UInt32 = try buf.readInt()
+        return Double(seconds) + (Double(microseconds) / 1.0e9)
+    }
+
+    fileprivate func writeDuration(into buf: Writer) {
+        if self.rounded(.down) > Double(Int64.max) {
+            fatalError("Duration overflow, exceeds max bounds supported by Uniffi")
+        }
+
+        if self < 0 {
+            fatalError("Invalid duration, must be non-negative")
+        }
+
+        let seconds = UInt64(self)
+        let microseconds = UInt32((self - Double(seconds)) * 1.0e9)
+        buf.writeInt(seconds)
+        buf.writeInt(microseconds)
```

Comment:
As above, the value written to the buffer should be named `nanoseconds`, not `microseconds`.